### ISSUE TITLE
New UI breaks when I'm running it in husky precommit

### ISSUE THREAD

**Issue by anthonyshew** [vercel/turbo#8311]
This new UI breaks when I'm running it in husky precommit:

In my root package.json:

```
    "lint": "turbo lint",
```

and in `.husky/pre-commit`:

```
pnpm lint
```

when running this, it gives:

```
$ git commit -m "Test commit"
• Packages in scope: ....
• Running lint in 8 packages
• Remote caching disabled
  × internal errors encountered: receiver dropped,receiver dropped,receiver dropped
```

The only way to fix this that I've found dso far is to add `"ui": "stream"` to the root `turbo.json`.

_Originally posted by @siawyoung in https://github.com/vercel/turbo/discussions/7802#discussioncomment-9672182_

**Comment by willsather** [comment]
We are facing this issue as well - seems like it's only occurring when any _one_ of our `pre-commit` commands is a cache miss, otherwise if all commands are a cache hit - it works fine.

Adding `"ui": "stream"` to the root `turbo.json` does solve the husky `pre-commit` issue, but unfortunately disables the new terminal UI.

<img width="570" alt="image" src="https://github.com/vercel/turbo/assets/56037657/8a23f19d-9643-4c64-abf6-65ab82e24b73">

**Comment by pieterjandebruyne** [comment]
Reproduce repo:
https://github.com/pieterjandebruyne/turbov2-husky-bug

### To Reproduce
run `git clone git@github.com:pieterjandebruyne/turbov2-husky-bug.git`
-> change any index.ts file
run `git add . && git commit -m "test"`

**Comment by pieterjandebruyne** [comment]
"ui": "stream" does fix the husky issue but I am getting different errors now when I use it inside of a dockerfile:

```
#19 ERROR: process "/bin/sh -c turbo run build --filter=frontend" did not complete successfully: exit code: 1
------
 > [installer 8/8] RUN turbo run build --filter=frontend:
0.207 Error: turbo_json_parse_error
0.207 
0.207   x Found an unknown value `Stream`.
0.207     ,-[57:1]
0.207  57 |   },
0.207  58 |   "ui": "Stream"
0.207     :         ^^^^^^^^
0.207  59 | }
0.207     `----
0.207 
------
Dockerfile:32
--------------------
  30 |     
  31 |     RUN npm -g i turbo@2.0.1
  32 | >>> RUN turbo run build --filter=frontend
  ```
  
Also not sure where the capitalised "Stream" string is coming from since in my turbo.json it's all lowercase

![Screenshot 2024-06-05 at 12 28 33](https://github.com/vercel/turbo/assets/9568274/a3d0b42d-2304-40de-95eb-676f62b656de)

`turbo run build --filter=frontend` outside of dicker is working fine...

**Comment by willsather** [comment]
Interesting - we also are running into `"ui": "Stream"`.  Our `turbo.json` file looks like:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": ["**/.env.*local"],
  "globalEnv": [
    "ENV VARS ..."
  ],
  "tasks": {
    "build": {
      "dependsOn": ["^build"],
      "outputLogs": "new-only"
    },
    "dev": {
      "cache": false,
      "persistent": true
    },
    "lint": {
      "outputLogs": "new-only"
    },
    "start": {
      "cache": false,
      "persistent": true
    },
    "test": {
      "outputLogs": "new-only"
    }
  },
  "ui": "stream"
}
```

but when we run `pnpm prune` (even outside of Docker) the `/out` folder shows the `turbo.json` as:

```json
{
  "$schema": "https://turbo.build/schema.json",
  "globalDependencies": ["**/.env.*local"],
  "globalEnv": [
    "ENV VARS ..."
  ],
  "tasks": {
    "build": {
      "dependsOn": ["^build"],
      "outputLogs": "new-only"
    },
    "dev": {
      "cache": false,
      "persistent": true
    },
    "lint": {
      "outputLogs": "new-only"
    },
    "start": {
      "cache": false,
      "persistent": true
    },
    "test": {
      "outputLogs": "new-only"
    }
  },
  "ui": "Stream"
}
```

which as mentioned above gives that error.  

<img width="577" alt="image" src="https://github.com/vercel/turbo/assets/56037657/432ffa56-7821-484a-aea8-1383ce2cffcf">


I can make a repo to reproduce later (or open a different issue), but it seems like this a completely different issue than the initial Husky problem... unfortunately though, means the `"ui": "stream"` workaround doesn't actually help us in this case.

**Comment by chris-olszewski** [comment]
@willsather @pieterjandebruyne 

Your issues with pre-commit failing should be addressed by https://github.com/vercel/turbo/pull/8333 which is in `2.0.2-canary.0`. If you could test it out that would be great! 

If you just want to disable the UI for commit hooks you could add [`TURBO_UI=0`](https://turbo.build/repo/docs/reference/system-environment-variables) as an env var to the commit hook. 

The `Found an unkown value "Stream"` error should be fixed with #8338

**Comment by chris-olszewski** [comment]
Both issues should be fixed in `2.0.2-canary.1` please test out and reopen if this isn't the case.
